### ISSUE TITLE
Rive crashes Unity editor

### ISSUE THREAD

**Issue by BitBarrel** [rive-app/rive-unity#5]
When adding the package from disk (via git fails, see other bug report), Rive will crash the Unity editor. I got it to work at some point, but now it crashes no matter what. It happens when using the RiveScreen.cs or RiveTexture.cs script.

I already set the renderer to OpenGL Core but it makes no difference.

Platform used is Windows, with an nVidia laptop GPU.

I also got this warning once, so maybe it is related:

NullReferenceException: Object reference not set to an instance of an object
Rive.RenderQueue.align (Rive.Fit fit, Rive.Alignment alignment, Rive.Artboard artboard) (at D:/Sources/Unity/rive-unity-0.1.16/package/Runtime/RenderQueue.cs:54)
Rive.RiveScreen.drawRive (Rive.RenderQueue queue) (at Assets/Rive/RiveScreen.cs:143)
Rive.RiveScreen.Start () (at Assets/Rive/RiveScreen.cs:138)

**Comment by luigi-rosso** [comment]
Hey @BitBarrel, sorry to hear about the woes. Is this using the basic example project or something else?

**Comment by BitBarrel** [comment]
The crash happens with all projects which contain the rive package and use it, both the example projects, and my own project.

The nullreferenceException only happens with my own project.

**Comment by luigi-rosso** [comment]
Thanks for the details! The various MonoBehaviour scripts are very much a work in progress/example implementation meant to show how to get started with using the Rive objects and the RenderQueue they'll likely not work on all render pipelines and versions of Unity, however they should definitely not cause crashes.

This helped us catch one of the issues:
```
System.EntryPointNotFoundException: unrefStateMachine assembly: type: member:(null)
at (wrapper managed-to-native) Rive.StateMachine.unrefStateMachine(intptr)
at Rive.StateMachine.Finalize () [0x00002] in D:\Sources\Unity\rive-unity-0.1.16\package\Runtime\StateMachine.cs:29
```

We'll have a fix up for that shortly. I'll ping when it's available as it'd be helpful if you could run it again and get us your Editor.log if you see more crashes.

Thanks again!

**Comment by luigi-rosso** [comment]
Does this still repro for you or has it been fixed @BitBarrel?

**Comment by BitBarrel** [comment]
It's fixed now, working well.

**Comment by schetle** [comment]
I'm receiving these crashes also.

Unity 2022.3.17f1 LTS (Latest LTS as of my posting of this)
Windows 11 22H2 22621.3007
rive-unity v0.1.53

DrawToCube scene from getting-started project. Dragging cute_robot to the cube's Rive Texture. Clicking play immediately crashes Unity Editor.

Using D3D11, Unity Editor crashes to desktop without the Crash Reporter.
Using OpenGLCore, Unity Editor crashes with a dump. A quick stack trace:

```========== OUTPUTTING STACK TRACE ==================

0x00007FFDABCE7A71 (rive) deleteEmbeddedAssetList
0x00007FFDABCE6C18 (rive) deleteEmbeddedAssetList
0x00007FFDABCE39F6 (rive) unrefArtboard
0x000001BE2BFBCFDA (Mono JIT Code) (wrapper managed-to-native) Rive.Artboard:unrefArtboard (intptr)
0x000001BE2BFBCEC3 (Mono JIT Code) [.\Library\PackageCache\app.rive.rive-unity@f2f5b2dafd\Runtime\Artboard.cs:29] Rive.Artboard:Finalize () 
0x000001BD9A7BD10C (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_virtual_void__this__ (object,intptr,intptr,intptr)
0x00007FFDAD306F57 (mono-2.0-bdwgc) [C:\build\output\Unity-Technologies\mono\mono\metadata\gc.c:383] mono_gc_run_finalize 
0x00007FFDAD58C928 (mono-2.0-bdwgc) [C:\build\output\Unity-Technologies\mono\external\bdwgc\finalize.c:1318] GC_invoke_finalizers 
0x00007FFDAD307FA3 (mono-2.0-bdwgc) [C:\build\output\Unity-Technologies\mono\mono\metadata\gc.c:929] finalize_domain_objects 
0x00007FFDAD308660 (mono-2.0-bdwgc) [C:\build\output\Unity-Technologies\mono\mono\metadata\gc.c:1036] finalizer_thread 
0x00007FFDAD29F98B (mono-2.0-bdwgc) [C:\build\output\Unity-Technologies\mono\mono\metadata\threads.c:1272] start_wrapper_internal 
0x00007FFDAD29FB9E (mono-2.0-bdwgc) [C:\build\output\Unity-Technologies\mono\mono\metadata\threads.c:1348] start_wrapper 
0x00007FFE68AF257D (KERNEL32) BaseThreadInitThunk
0x00007FFE6A1EAA58 (ntdll) RtlUserThreadStart

========== END OF STACKTRACE ===========```

**Comment by luigi-rosso** [comment]
We've got a fix for the D3D11 issue coming soon. The RenderTexture is required to have enableRandomWrite set to true when using D3D11. We're shortly going to remove this restriction but for now this should unblock you. In the meantime we're putting in a fix that warns you if you try to use a RenderTexture that doesn't have this set.

The OpenGLCore issue is more mysterious, does it happen right when pressing play? I have a very similar setup to yours but I can't repro this.

**Comment by luigi-rosso** [comment]
The D3D11 issue should warning users when enableRandomWrite isn't set is in [v0.1.56](https://github.com/rive-app/rive-unity/releases/tag/v0.1.56)

**Comment by schetle** [comment]
> We've got a fix for the D3D11 issue coming soon. The RenderTexture is required to have enableRandomWrite set to true when using D3D11. We're shortly going to remove this restriction but for now this should unblock you. In the meantime we're putting in a fix that warns you if you try to use a RenderTexture that doesn't have this set.
> 
> The OpenGLCore issue is more mysterious, does it happen right when pressing play? I have a very similar setup to yours but I can't repro this.

For OpenGLCore:
Crash type #1 crashes instantly once Play is hit.
Crash type #2 crashes as soon as you click a rive asset in the Unity content browser.

**Comment by damzobridge** [comment]
This should be fixed from [v0.1.188](https://github.com/rive-app/rive-unity/releases/tag/v0.1.188)! Let us know if you still run into any crashes.
